selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/stack/CreateWorkspaceFromCeylonWithJavaStackTest.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.selenium.stack;
+
+import static org.eclipse.che.commons.lang.NameGenerator.generate;
+import static org.eclipse.che.selenium.core.constant.TestProjectExplorerContextMenuConstants.ContextMenuCommandGoals.BUILD_GOAL;
+import static org.eclipse.che.selenium.core.constant.TestProjectExplorerContextMenuConstants.ContextMenuCommandGoals.RUN_GOAL;
+import static org.eclipse.che.selenium.pageobject.dashboard.NewWorkspace.Stack.CEYLON_WITH_JAVA_JAVASCRIPT;
+
+import com.google.inject.Inject;
+import org.eclipse.che.selenium.core.client.TestWorkspaceServiceClient;
+import org.eclipse.che.selenium.core.user.DefaultTestUser;
+import org.eclipse.che.selenium.pageobject.Consoles;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.eclipse.che.selenium.pageobject.dashboard.CreateWorkspaceHelper;
+import org.eclipse.che.selenium.pageobject.dashboard.Dashboard;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+/** @author Skoryk Serhii */
+public class CreateWorkspaceFromCeylonWithJavaStackTest {
+
+  private static final String WORKSPACE_NAME = generate("workspace", 4);
+  private static final String PROJECT_NAME = "ceylon-hello-world";
+  private static final String MODULE_COMPILED_MESSAGE =
+      "Note: Created module che.ceylon.samples.helloWorld";
+  private static final String MODULE_STARTED_MESSAGE =
+      "Hello World from Ceylon on the following backend : ";
+
+  @Inject private Ide ide;
+  @Inject private Consoles consoles;
+  @Inject private Dashboard dashboard;
+  @Inject private DefaultTestUser defaultTestUser;
+  @Inject private ProjectExplorer projectExplorer;
+  @Inject private CreateWorkspaceHelper createWorkspaceHelper;
+  @Inject private TestWorkspaceServiceClient workspaceServiceClient;
+
+  @BeforeClass
+  public void setUp() {
+    dashboard.open();
+  }
+
+  @AfterClass
+  public void tearDown() throws Exception {
+    workspaceServiceClient.delete(WORKSPACE_NAME, defaultTestUser.getName());
+  }
+
+  @Test
+  public void checkWorkspaceCreationFromCeylonWithJavaStack() {
+    createWorkspaceHelper.createWorkspaceFromStackWithProject(
+        CEYLON_WITH_JAVA_JAVASCRIPT, WORKSPACE_NAME, PROJECT_NAME);
+
+    ide.switchToIdeAndWaitWorkspaceIsReadyToUse();
+
+    projectExplorer.waitProjectInitialization(PROJECT_NAME);
+  }
+
+  @Test(priority = 1)
+  public void checkCeylonHelloWorldProjectCommands() {
+    consoles.executeCommandFromProjectExplorer(
```

Comment:
It's not quite clear what next sequences of command are doing. Could you, please, comment it in the code?